lib/semian.rb

```
@@ -166,6 +167,32 @@ def destroy(name)
   def resources
     @resources ||= {}
   end
+
+  private
+
+  def create_circuit_breaker(name, **options)
+    circuit_breaker = options.fetch(:circuit_breaker, true)
+    return unless circuit_breaker
+
+    exceptions = options[:exceptions] || []
+    CircuitBreaker.new(
+      name,
+      success_threshold: options[:success_threshold],
+      error_threshold: options[:error_threshold],
+      error_timeout: options[:error_timeout],
+      exceptions: Array(exceptions) + [::Semian::BaseError],
+      implementation: ::Semian::Simple,
+    )
+  end
+
+  def create_bulkhead(name, **options)
+    bulkhead = options.fetch(:bulkhead, true)
+    return unless bulkhead
+
+    permissions = options[:permissions] || 0660
+    timeout = options[:timeout] || 0
+    Resource.new(name, tickets: options[:tickets], permissions: permissions, timeout: timeout)
```

Same thing goes here, we need to raise if `tickets` are not defined.